javers-core/src/main/java/org/javers/core/metamodel/scanner/PropertyScanner.java

```diff
@@ -5,6 +5,14 @@
  *
  * @author pawel szymczyk
  */
-interface PropertyScanner {
-    PropertyScan scan(Class<?> managedClass);
+abstract class PropertyScanner {
+    abstract public PropertyScan scan(Class<?> managedClass, boolean ignoreDeclaredProperties);
+
+    public PropertyScan scan(Class<?> managedClass, ClassAnnotationsScan classScan) {
```

Comment:
I'm not sure if PropertyScanner should be aware of ClassAnnotationsScan. Is it not an unnecessary coupling? The only information needed here is whether declared properties should be ignored.